distutils/dist.py

```diff
@@ -1189,7 +1191,10 @@ def get_version(self):
         return self.version or "0.0.0"
 
     def get_fullname(self):
-        return f"{self.get_name()}-{self.get_version()}"
+        return "{}-{}".format(
+            canonicalize_name(self.get_name()).replace('-', '_'),
+            self.get_version(),
```

Comment:
Maybe something  like https://github.com/pypa/setuptools/pull/4434

Reply:
Thanks for that! Done in 57cab7118.